### PR TITLE
Remove spylon-kernel from all images.

### PR DESCRIPTION
**Pull request by bjornjorgensen**
## Describe your changes
This is part 1 of 2 for [updating Apache Spark to 3.3.0 with Scala 2.13 and Java version 17](https://github.com/jupyter/docker-stacks/pull/1727).
This PR will remove the spylon-kernel, which means that there will be no more Scala support.
The Spylon-kernel which was last updated in 2017 and is inactive. 
When updating to Spark 3.3.0 with Scala 2.13, tests for the spylon kernel are failing. 
[100%] FAILED tests/all-spark-notebook/test_spark_notebooks.py::test_nbconvert[local_spylon] 

When we upgrade pyspark (python on spark) image, the tests are running for the all-spark-notebook. The all-spark-notebook are pyspark, R and Scala. Scala is provided thru the not updated spylon-kernel, and this is the reason why we need to remove it.
The only reason why we have the spylon kernel is to have support for notebooks with Scala to use Spark.

However Apache Spark have now [voted](https://lists.apache.org/thread/3fd2n34hlyg872nr55rylbv5cg8m1556) to make a [client](https://docs.google.com/document/d/1Mnl6jmGszixLW4KcJU5j9IgpG9-UabS0dcM6PM2XGDc/edit) [The JIRA issue](https://issues.apache.org/jira/browse/SPARK-39375)
So for the next Apache Spark 3.4.0 this issue will hopefully be solved by the Spark Client.  

## Issue ticket if applicable

[Upgrading Spark->3.3,Hadoop->3,Scala->2.13,Java->17](https://github.com/jupyter/docker-stacks/pull/1727)

## Checklist (especially for first-time contributors)

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [x] I will try not to use force-push to make the review process easier for reviewers
- [x] I have updated the documentation for significant changes

### CODE REVIEW

**Comment by bjornjorgensen** [comment]
@mathbunnyru When this PR is merged I will start working on a new PR to upgrade to spark 3.3.0

File: docs/using/selecting.md [hunk 1]
```diff
@@ -183,7 +183,6 @@ communities.
   [sparklyr](https://spark.rstudio.com),
   [ggplot2](https://ggplot2.tidyverse.org)
   packages
-- [spylon-kernel](https://github.com/vericast/spylon-kernel) to support Scala code in Jupyter notebooks
```
Comment (mathbunnyru) [inline docs/using/selecting.md:186]: A few lines above there is still mention of `Scala`.
Please, make sure, there is no `scala` or `spylon` words in our codebase.

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

**Comment by mathbunnyru** [comment]
In general, this LGTM 👍 
@consideRatio could you take a look, please?

When we get rid of leftovers, I would probably like to squash-merge this, to make it easier to see what needs to be added back in the future (if another kernel will be added).

**Review by mathbunnyru — APPROVED** [review]

**Comment by consideRatio** [comment]
I'm not confident on the situation overall, entirely lacking experience with spark, but this looks reasonable to me.

I suggest:
- updating the title to clarify this relates to the "all spark" image
- clarifying the assumption currently being hidden: that this image is meant to support work against spark or similar, and that what is removed here is justified based on that.
- don't assume knowledge that the spylon kernel is coupled with scala (which I've just guessed it is)

**Comment by bjornjorgensen** [comment]
When updating to Spark 3.3.0 with Scala 2.13 [Test for spylon kernel are failing.](https://github.com/jupyter/docker-stacks/runs/6924663413?check_suite_focus=true) [100%] FAILED tests/all-spark-notebook/test_spark_notebooks.py::test_nbconvert[local_spylon]
The [Spylon-kernal](https://github.com/vericast/spylon-kernel)  witch wos last updated in 2017 and is inactive. 

The [Ammonite-spark](https://github.com/alexarchambault/ammonite-spark) have support for Scala 2.13 It then run Spark by using [ivy](https://ant.apache.org/ivy/)

[Toree](https://toree.apache.org) is another one for Spark and Scala. This got support for  [java version 8 and 11.](https://github.com/apache/incubator-toree/pull/194) We are going to upgrade to 17 now.  

There are something that tells me that there are something wrong in with the CI 
WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/usr/local/spark-3.2.1-bin-hadoop3.2/jars/spark-unsafe_2.12-3.2.1.jar) to constructor java.nio.DirectByteBuffer(long,int)
This means that we are upgrading pyspark image, but the tests running for the all-spark-notebook are using old images. 

I have put this PR in WIP mode so that we can find a proper solution to this problem.

**Comment by consideRatio** [comment]
Thanks!

Btw, I think maybe avoiding adding something else for scala is the best course of action until its a clear wish from an active user of these images. Adding it before that seems premature.

**Comment by bjornjorgensen** [comment]
Yes, but when we upgrade pyspark (python on spark) image, the tests are running for the all-spark-notebook. all-spark-notebook are pyspark, r and scala. Scala is provided thru the not updated spylon-kernel, and this is the problem. 

Now I don't really wanna remove support for Scala in all images, but it's tested by spark.

**Comment by bjornjorgensen** [comment]
@consideRatio Now I have updated the description for this PR. Are there still any questions?

**Comment by bjornjorgensen** [comment]
@Bidek56 Yes, but at the moment there are no Scala kernels that have support for Spark 3.3.X with Java 17 and Scala 2.13. That's the reason why [your PR failed to build now](https://github.com/jupyter/docker-stacks/runs/6991536832?check_suite_focus=true#step:6:1291).. 

You can link to the Almond kernel, but it wont help users that will run the latest Spark version. We can apply this PR which removes Scala. And when the Almond kernel gets support for the latest Spark version, we can make a new PR with Scala support through the Almond kernel.

**Comment by mathbunnyru** [comment]
> Thanks!
> 
> Btw, I think maybe avoiding adding something else for scala is the best course of action until its a clear wish from an active user of these images. Adding it before that seems premature.

I agree with @consideRatio.
Let's remove `spylon-kernel` for now because we're not even sure people use scala that much in our images.
Also, this kernel is deprecated and stops us from moving forward, and this is something our users clearly want.

When another kernel works with our images, we will try to document it or merge it into our images.

Merging this one, thank you for your contribution 👍

**Comment by mathbunnyru** [comment]
Squash merging this to allow easier reverts if needed.

**Pull request merged**